WireDomain/Package.swift

```diff
@@ -7,24 +7,44 @@ let package = Package(
     name: "WireDomainPackage",
     platforms: [.iOS("16.4"), .macOS(.v12)],
     products: [
-        .library(name: "WireDomainPackage", targets: ["WireDomainPkg"])
+        .library(name: "WireDomainPackage", targets: ["WireDomainPackage"]),
+        .library(name: "WireDomainPackageSupport", targets: ["WireDomainPackageSupport"])
     ],
     dependencies: [
-        .package(url: "https://github.com/swiftlang/swift-docc-plugin", from: "1.1.0")
+        .package(url: "https://github.com/swiftlang/swift-docc-plugin", from: "1.1.0"),
+        .package(path: "../WireAPI"),
+        .package(path: "../WireFoundation"),
+        .package(path: "../WireLogging"),
+        .package(path: "../WirePlugins"),
     ],
     targets: [
         .target(
-            name: "WireDomainPkg",
-            path: "./Sources/WireDomain",
-            sources: ["./UseCases/Protocols"]
-        )
```

Comment:
Two targets shared the same directory tree: "WireDomain" (Xcode project) and "WireDomainPkg" (in Swift package).
Now the the files of "WireDomainPackage" are moved to their own subtree.

Reply:
question: why do we need the WireDomainPackage and Project again? It always confuse me. I wonder if we went to quickly on moving to SPM and have redundancy here

Reply:
Because some code in WireDomain depends on WireSyncEngine and the other Xcode projects, while some code of WireDomain is already used by Swift packages (WireUI etc.).

We might have adopted Swift Packages too early, I agree.
But there are also some benefits of swift packages we wouldn't want to miss now, like the `package` access modifier.